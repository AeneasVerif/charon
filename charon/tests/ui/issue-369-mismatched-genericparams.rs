pub trait Try: FromResidual<()> {
    type Residual;
}

pub trait FromResidual<R> {}

impl<T> Try for Option<T> {
    type Residual = ();
}

impl<T> FromResidual<<Self as Try>::Residual> for Option<T> {}
