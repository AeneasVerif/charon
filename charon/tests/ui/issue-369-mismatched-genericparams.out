# Final LLBC before serialization:

// Full name: core::marker::MetaSized
#[lang_item("meta_sized")]
pub trait MetaSized<Self>

// Full name: core::marker::Sized
#[lang_item("sized")]
pub trait Sized<Self>
{
    parent_clause0 : [@TraitClause0]: MetaSized<Self>
    non-dyn-compatible
}

// Full name: core::option::Option
#[lang_item("Option")]
pub enum Option<T>
where
    [@TraitClause0]: Sized<T>,
{
  None,
  Some(T),
}

// Full name: test_crate::FromResidual
pub trait FromResidual<Self, R>
{
    parent_clause0 : [@TraitClause0]: MetaSized<Self>
    parent_clause1 : [@TraitClause1]: Sized<R>
    vtable: test_crate::FromResidual::{vtable}<R>
}

// Full name: test_crate::Try
pub trait Try<Self>
{
    parent_clause0 : [@TraitClause0]: MetaSized<Self>
    parent_clause1 : [@TraitClause1]: FromResidual<Self, ()>
    parent_clause2 : [@TraitClause2]: Sized<Self::Residual>
    type Residual
    non-dyn-compatible
}

// Full name: test_crate::{impl FromResidual<()> for Option<T>[@TraitClause0]}
impl<T> FromResidual<()> for Option<T>[@TraitClause0]
where
    [@TraitClause0]: Sized<T>,
{
    parent_clause0 = MetaSized<Option<T>[@TraitClause0]>
    parent_clause1 = Sized<()>
    vtable: {impl FromResidual<()> for Option<T>[@TraitClause0]}::{vtable}<T>[@TraitClause0]
}

// Full name: test_crate::{impl Try for Option<T>[@TraitClause0]}
impl<T> Try for Option<T>[@TraitClause0]
where
    [@TraitClause0]: Sized<T>,
{
    parent_clause0 = MetaSized<Option<T>[@TraitClause0]>
    parent_clause1 = {impl FromResidual<()> for Option<T>[@TraitClause0]}<T>[@TraitClause0]
    parent_clause2 = Sized<()>
    type Residual = ()
    non-dyn-compatible
}



